inc/saifdb.h

```diff
@@ -178,6 +178,16 @@ typedef enum _sai_fdb_entry_attr_t
      */
     SAI_FDB_ENTRY_ATTR_COUNTER_ID,
 
+    /**
+     * @brief Specifies whether a MAC move is allowed
+     *
+     * @type bool
+     * @flags CREATE_AND_SET
+     * @default false
+     * @validonly SAI_FDB_ENTRY_ATTR_TYPE == SAI_FDB_ENTRY_TYPE_STATIC
+     */
+    SAI_FDB_ENTRY_ATTR_ALLOW_MAC_MOVE,
```

Comment:
Can you elaborate on the use-case where we want to allow MAC_MOVE of a static MAC address?

Reply:
As discussed in the SAI community meeting on Dec 19 the use case is for EVPN wherein the MACs donot age out as they are created and deleted by the EVPN control plane (BGP) . In that respect they are static MACs. However these MACs must be subjected to a MAC Move. For e.g. a MAC installed by BGP moves from the remote to a local port. This should trigger a hardware learn event. The SAI attribute is to allow for the hardware to learn the MAC even though the MAC is programmed as static.

Reply:
can you add some clarification on the TRAP attributed introduced in #696 ? What is the expectation of that notification when MAC_MOVE is enabled or disabled on this attribute.

Reply:
The trap introduced in #696 is a generic trap, where as this PR is about a specific MAC entry. So the specific entry would take precedence over a generic trap. I hope this clarifies.

Reply:
(Additional clarification) When MAC_MOVE is explicitly disabled for a static mac-entry, the trap (introduced in #696) would also not be generated.

Reply:
Is this also expected to work with MLAG?

The behavior of this entry is closer to being "dynamic but not ageable" than "static but movable" because static FDB is not supposed to be changed by anything besides the application that created it, whereas dynamic is expected to change, and we just say that it does not expire. Let me know if you considered this way of definition.

Reply:
Yes this is expected to work with MCLAG as well. Please refer to the MC Lag enhancements HLD floated in sonic forum. Static vs dynamic refers to a MAC being ageable and not whether they are movable. Here we are adding an attribute to static mac specifying if this is moveable or not.

Reply:
@bandaru-viswanath, thanks for the clarification. can you make the clarification in the header?

Reply:
@lguohan Updated the header file.